be/src/olap/rowset/segment_v2/column_writer.cpp

```diff
@@ -226,10 +228,14 @@ Status ColumnWriter::_write_data_page(Page* page) {
 Status ColumnWriter::_write_physical_page(std::vector<Slice>* origin_data, PagePointer* pp) {
     std::vector<Slice>* output_data = origin_data;
     std::vector<Slice> compressed_data;
-    // TODO(zc): support compress
-    // if (_need_compress) {
-    //     output_data = &compressed_data;
-    // }
+
+    // Put compressor out of if block, because we should use compressor's
+    // content until this function finished.
+    PageCompressor compressor(_compress_codec);
+    if (_compress_codec != nullptr) {
+        RETURN_IF_ERROR(compressor.compress(*origin_data, &compressed_data));
+        output_data = &compressed_data;
+    }
 
     // checksum
     uint8_t checksum_buf[sizeof(uint32_t)];
```

Comment:
if _opts.need_checksum is false, checksum_buf will be leaked.

Reply:
this is a stack array, can not leak